readalongs/web_api.py

```diff
@@ -89,21 +93,20 @@ class AssembleResponse(BaseModel):
     g2ped: Optional[str]
 
 
-def process_xml(func):
-    # Wrapper for processing XML, reads the XML with proper encoding,
-    # then applies the given function to it,
-    # then converts the result back to utf-8 XML and returns it
-    def wrapper(xml, **kwargs):
-        parsed = etree.fromstring(bytes(xml.xml, encoding=xml.encoding))
-        processed = func(parsed, **kwargs)
-        return etree.tostring(processed, encoding="utf-8", xml_declaration=True)
-
-    return wrapper
-
-
 @v1.get("/langs", response_model=Dict[str, str])
 async def langs() -> Dict[str, str]:
-    """Return the list of supported languages and their names as a dict."""
+    """Return the list of supported languages and their names as a dict.
+
+    Returns:
+        langs as dict with language codes as keys and the full language name as
+        values, e.g.:
+        `{
+            "alq", "Algonquin",
+            "atj": "Atikamekw",
+            "lc3", "Third Language Name",
+            ...
+        }`
+    """
```

Comment:
Ideally this, and all of the removal of encoding parameters, would be a separate PR, but git doesn't make that very easy to do in my experience...

Reply:
Actually, what makes it hard is that my changes are kinda mixed together. If I had decided to separate the two tasks, I could have easily done it, and I guess I should have. Git actually helps doing that, but I was just too lazy to do it. Full disclosure: I consider myself a Git wizard and there's not much I can't do with it. I think it would have been easy, but a "normal" person would probably disagree. :)